cli/azd/cmd/util.go

```diff
@@ -453,16 +454,17 @@ func promptTemplate(ctx context.Context, message string, askOne Asker) (string,
 		Options: templateNames,
 		Default: templateNames[0],
 	}, &selectedTemplateIndex); err != nil {
-		return "", fmt.Errorf("prompting for template: %w", err)
+		return result, fmt.Errorf("prompting for template: %w", err)
 	}
 
 	if selectedTemplateIndex == 0 {
-		return "", nil
+		return result, nil
 	}
 
-	log.Printf("Selected template: %s", fmt.Sprint(templateNames[selectedTemplateIndex]))
+	selectedTemplateName := templateNames[selectedTemplateIndex]
+	log.Printf("Selected template: %s", fmt.Sprint(selectedTemplateName))
 
-	return templateNames[selectedTemplateIndex], nil
+	return templatesSet[selectedTemplateName], nil
```

Comment:
@ellismg , I updated using a simple array of templates to have a set where we can use the key for constant access instead of the lineal search